packages/react/src/FormattingToolbar/components/DefaultDropdowns/BlockTypeDropdown.tsx

```diff
@@ -10,7 +15,38 @@ import {
 } from "react-icons/ri";
 import { ToolbarDropdown } from "../../../SharedComponents/Toolbar/components/ToolbarDropdown";
 
-export const BlockTypeDropdown = (props: { editor: BlockNoteEditor }) => {
+const arrayEquals = (a: readonly string[], b: readonly string[]) => {
+  if (a.length !== b.length) {
+    return false;
+  }
+
+  for (let i = 0; i < a.length; i++) {
+    if (a[i] !== b[i]) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+const shouldShow = (schema: BlockSchema) => {
```

Comment:
It's not clear to me why we use the `arrayEquals` on schema.

Wouldn't it make more sense to just show a dropdown for all the levels in the propschema?